contracts/Core/AssetManager.sol

```diff
@@ -90,13 +91,25 @@ contract AssetManager is ACL, AssetStorage, Constants, StateManager {
 
             emit JobActivityStatus(jobs[id].active, id, epoch, block.timestamp);
         } else {
-            collections[id].active = assetStatus;
             if (assetStatus) {
-                numActiveAssets = numActiveAssets + 1;
+                if (!collections[id].active) {
+                    activeAssets.push(id);
+                    collections[id].assetIndex = uint8(activeAssets.length);
+                }
             } else {
-                numActiveAssets = numActiveAssets - 1;
+                if (collections[id].active) {
+                    uint8 assetIndex = collections[id].assetIndex;
+                    activeAssets[assetIndex - 1] = activeAssets[activeAssets.length - 1];
+                    activeAssets.pop();
+                    collections[id].assetIndex = 0;
+                    if ((assetIndex - 1) != activeAssets.length) {
```

Comment:
hey rather than having if block here cant we have it on top only, like if its last element then only pop, if its not then swap.

Reply:
Done